pkg/controller/provider/infoblox/access.go

```diff
@@ -82,12 +82,15 @@ func (this *access) NewRecord(fqdn string, rtype string, value string, zone prov
 		if n, err := strconv.Unquote(value); err == nil && !strings.Contains(value, " ") {
 			value = n
 		}
-		return (*RecordTXT)(ibclient.NewRecordTXT(ibclient.RecordTXT{
+		record = (*RecordTXT)(ibclient.NewRecordTXT(ibclient.RecordTXT{
 			Name: fqdn,
 			Text: value,
 			//Zone: zone.Key(),
 			View: this.view,
 		}))
 	}
-	return nil
+	if record != nil {
+		record.SetTTL(int(ttl))
+	}
+	return
```

Comment:
Hi Martin, if you extract the setting of the ttl anyway, you could also extract the other common settings (Name and View)

And you could put the common record creation at the beginning and just set the dedicated info field in the cases and directly return the record. At the end then the simple nil return could be kept and the named return value could easily be avoided.